tests/test_project/CopyModulesWebpackPlugin.test.js

```diff
@@ -86,4 +90,75 @@ describe('copy-modules-webpack-plugin', function() {
 
     testPluginConfig(pluginConfig, pathsThatShouldBeCopied, done);
   });
+
+  it('copies relevant package.json files in addition to javascript files when includePackageJsons is true',
+      function(done) {
+        const pathsThatShouldBeCopied = [
+              'mock_src/b.js',
+              'mock_src/e.js',
+              'mock_src/subfolder/a.js',
+              'package.json',
+              'node_modules/foo-pkg/foo.js',
+              'node_modules/foo-pkg/package.json',
+              'node_modules/bar-pkg/bar.js',
+              'node_modules/bar-pkg/package.json',
+              '__..__/node_modules/outside-cwd-pkg/outside-cwd.js',
+              '__..__/node_modules/outside-cwd-pkg/package.json'
+            ],
+            pluginConfig = { includePackageJsons: true };
+
+        testPluginConfig(pluginConfig, pathsThatShouldBeCopied, done);
+      }
+  );
+
+  it('does not copy package.json files when includePackageJsons is false', function(done) {
+    const pathsThatShouldBeCopied = [
+          'mock_src/b.js',
+          'mock_src/e.js',
+          'mock_src/subfolder/a.js',
+          'node_modules/foo-pkg/foo.js',
+          'node_modules/bar-pkg/bar.js',
+          '__..__/node_modules/outside-cwd-pkg/outside-cwd.js'
+        ],
+        pluginConfig = { includePackageJsons: false };
+
+    testPluginConfig(pluginConfig, pathsThatShouldBeCopied, done);
+  });
+
+  it('handles files that have no parent package.json', function(done) {
+    const tmpDir = tmp.dirSync(),
+        tmpDirName = tmpDir.name;
+
+    tmpDirs.push(tmpDir);
+
+    // copy the js files but not the package.json files to a temp dir where (hopefully) there isn't a package.json
+    // anywhere higher in the directory tree
```

Comment:
I don't understand why copying the mock src project to another dir is necessary to execute this test case - but then again, maybe I don't understand what you're testing for.

I'm wondering if maybe there is a simpler way to set up the test data. (As a side note, the `(hopefully)` in this comment does not inspire confidence :)) Would it not be sufficient to add another subfolder to `mock_src` (and to the associated `node_modules` dir it contains) that does not have an associated `package.json` file? Or potentially just set up a second `mock_src` (say, `mock_src_sans_packageJson_files`) and test on that?

Reply:
> I'm wondering if maybe there is a simpler way to set up the test data. (As a side note, the (hopefully) in this comment does not inspire confidence :)) Would it not be sufficient to add another subfolder to mock_src (and to the associated node_modules dir it contains) that does not have an associated package.json file? Or potentially just set up a second mock_src (say, mock_src_sans_packageJson_files) and test on that?

I was going to do that but then I realized the `package.json` in the root of this repo would get picked up.  The only way to test the case where no parent dir at all has a package.jso is to do so from completely outside of the repo.  Or, technically, we could move the whole project down into a subdir and then have some test data in a sibling dir, but setting up this test the way that I have felt less invasive than that.

Reply:
> (As a side note, the (hopefully) in this comment does not inspire confidence :))

The reason for that is that we can't control what's in the filesystem outside of this project.  Aside from trying to make a `chroot` (which I don't believe is possible on Windows), I don't think there's a way around that

Reply:
> > I'm wondering if maybe there is a simpler way to set up the test data. [...]
> 
> I was going to do that but then I realized the `package.json` in the root of this repo would get picked up. The only way to test the case where no parent dir at all has a package.json is to do so from completely outside of the repo. Or, technically, we could move the whole project down into a subdir and then have some test data in a sibling dir, but setting up this test the way that I have felt less invasive than that.

Hmm, makes sense, I did not consider the package.json at the root of this project. Thanks for the explanation! I agree that your solution is less invasive than re-structuring the entire project.